frontend/packages/console-app/src/components/nodes/NodesPage.tsx

```diff
@@ -40,60 +56,108 @@ const NodeTableHeader = () => {
+    },
   ];
 };
 NodeTableHeader.displayName = 'NodeTableHeader';
 
-const NodesTableRow: React.FC<NodesTableRowProps> = ({ obj: node, index, key, style }) => {
-  const { name: machineName, namespace: machineNamespace } = getNodeMachineNameAndNamespace(node);
-  const nodeName = getName(node);
-  const nodeUID = getUID(node);
-  return (
-    <TableRow id={nodeUID} index={index} trKey={key} style={style}>
-      <TableData className={tableColumnClasses[0]}>
-        <ResourceLink kind={referenceForModel(NodeModel)} name={nodeName} title={nodeUID} />
-      </TableData>
-      <TableData className={tableColumnClasses[1]}>
-        <NodeStatus node={node} />
-      </TableData>
-      <TableData className={tableColumnClasses[2]}>
-        <NodeRoles node={node} />
-      </TableData>
-      <TableData className={tableColumnClasses[3]}>
-        {machineName && (
-          <ResourceLink
-            kind={referenceForModel(MachineModel)}
-            name={machineName}
-            namespace={machineNamespace}
-          />
-        )}
-      </TableData>
-      <TableData className={tableColumnClasses[4]}>
-        <ResourceKebab actions={menuActions} kind={referenceForModel(NodeModel)} resource={node} />
-      </TableData>
-    </TableRow>
-  );
+const mapStateToProps = ({ UI }) => ({
+  metrics: UI.getIn(['metrics', 'node']),
+});
+
+type NodesRowMapFromStateProps = {
+  metrics: NodeMetrics;
 };
+
+const NodesTableRow = connect<NodesRowMapFromStateProps, null, NodesTableRowProps>(mapStateToProps)(
+  ({ obj: node, index, key, style, metrics }: NodesTableRowProps & NodesRowMapFromStateProps) => {
+    const nodeName = getName(node);
+    const nodeUID = getUID(node);
+    const freeMem: number = Number(_.get(metrics, ['freeMemory', nodeName]));
+    const totalMem: number = Number(_.get(metrics, ['totalMemory', nodeName]));
+    const usedMemory: string = humanizeBinaryBytes(totalMem - freeMem).string;
+    const totalMemory: string = humanizeBinaryBytes(totalMem).string;
+    const cores: number = Number(_.get(metrics, ['cpu', nodeName]));
+    const freeStrg: number = Number(_.get(metrics, ['freeStorage', nodeName]));
+    const totalStrg: number = Number(_.get(metrics, ['totalStorage', nodeName]));
+    const usedStorage: string = humanizeBinaryBytes(totalStrg - freeStrg).string;
+    const totalStorage: string = humanizeBinaryBytes(totalStrg).string;
+    return (
+      <TableRow id={nodeUID} index={index} trKey={key} style={style}>
+        <TableData className={tableColumnClasses[0]}>
+          <ResourceLink kind={referenceForModel(NodeModel)} name={nodeName} title={nodeUID} />
+        </TableData>
+        <TableData className={tableColumnClasses[1]}>
+          <NodeStatus node={node} />
+        </TableData>
+        <TableData className={tableColumnClasses[2]}>
+          <NodeRoles node={node} />
+        </TableData>
+        <TableData className={tableColumnClasses[3]}>{node.status?.capacity?.pods}</TableData>
```

Comment:
This is confusing to me. We're showing the capacity for pods, but not the actual number of pods on the node? I think we should change the column header if that's what we really mean to show.

Reply:
Thanks for pointing this out. I need to take data from prom and use it here.